pkg/models/host.go

```diff
@@ -17,6 +25,158 @@ type Host struct {
 	SSHCommonArgs     string
 }
 
+const (
+	shell     = "/bin/bash"
+	localhost = "127.0.0.1"
+)
+
+// GetNodeID returns the node ID of the host.
+//
+// It checks if the node ID has a prefix of constants.AWSNodeAnsiblePrefix
+// and removes the prefix if present. Otherwise, it joins the first two parts
+// of the node ID split by "_" and returns the result.
+//
+// Returns:
+//   - string: The node ID of the host.
+func (h Host) GetCloudID() string {
+	if strings.HasPrefix(h.NodeID, constants.AWSNodeAnsiblePrefix) {
+		return strings.TrimPrefix(h.NodeID, constants.AWSNodeAnsiblePrefix+"_")
+	} else if strings.HasPrefix(h.NodeID, constants.GCPNodeAnsiblePrefix) {
+		return strings.TrimPrefix(h.NodeID, constants.GCPNodeAnsiblePrefix+"_")
+	}
+	// default behaviour - TODO refactor for other clouds
+	return strings.Join(strings.Split(h.NodeID, "_")[:2], "_")
+}
+
+// Connect starts a new SSH connection with the provided private key.
+//
+// It returns a pointer to a goph.Client and an error.
+func (h Host) Connect(timeout time.Duration) (*goph.Client, error) {
+	if timeout == 0 {
+		timeout = constants.SSHScriptTimeout
+	}
+	// Start new ssh connection with private key.
+	auth, err := goph.Key(h.SSHPrivateKeyPath, "")
+	if err != nil {
+		return nil, err
+	}
+	client, err := goph.NewConn(&goph.Config{
+		User:    h.SSHUser,
+		Addr:    h.IP,
+		Port:    22,
+		Auth:    auth,
+		Timeout: timeout,
+		// #nosec G106
+		Callback: ssh.InsecureIgnoreHostKey(),
+	})
+	if err != nil {
+		return nil, err
+	}
+	return client, nil
+}
+
+// Upload uploads a local file to a remote file on the host.
+func (h Host) Upload(localFile string, remoteFile string) error {
+	client, err := h.Connect(constants.SSHFileOpsTimeout)
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+	return client.Upload(localFile, remoteFile)
```

Comment:
I can but it will waste of time if folder exists and waste of time to check if folder exists for each upload ( making things slower), so I leave it to user to make sure he knows what he is doing